lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -416,7 +416,11 @@ TypeChecker::typeCheckTarget(SolutionApplicationTarget &target,
   // Construct a constraint system from this expression.
   ConstraintSystemOptions csOptions = ConstraintSystemFlags::AllowFixes;
 
-  if (DiagnosticSuppression::isEnabled(Context.Diags))
+  /// If we have a SolutionCallback, we are inspecting constraint system
+  /// solutions directly and thus also want to receive ambiguous solutions.
```

Comment:
This is actually interesting because there could be a number of _valid_ solutions produced by the solver and I'm guessing we don't really want to run `salvage` in that case, right? It feels like this check needs to be embedded into the solver itself somehow...

Reply:
You mean that we should just allow fixes straight away like we do for code completion? 

I deliberately wanted to keep this as close to the normal type checking as possible (without all the code-completion specific rules of weaker filtering etc.) because if normal type checking reaches a single, definite result, we also want to report that as the single result for cursor info (for now). Also, in contrast to code completion, there is a reasonable chance that the code is in a valid state and thus it makes sense to run normal type-checking first and only salvage if we didn’t find a solution or the solutions were ambiguous.

Reply:
I added this check to `ConstraintSystem::solve` now. Could you take another look at it?